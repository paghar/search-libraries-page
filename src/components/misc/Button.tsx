interface IBtnProps{   
  id?: string,
  type:"button" | "submit" | "reset",
  className?:string,
  onClick:(event: any)=>void,
  children?: React.ReactNode,
  text?:string 
}
  
const Button = (props:IBtnProps) => {
  return (
    <button 
      id={props.id}
      type={props?.type}       
      onClick={props?.onClick}  
      className={`${props.className}
        inline-flex items-center 
        py-2.5 px-3 ml-2 text-sm 
        font-medium text-white
        bg-pink-700 rounded-lg border border-pink-700
        hover:bg-pink-800 focus:ring-4 
        focus:outline-none focus:ring-pink-300
        dark:bg-pink-600 dark:hover:bg-pink-700 dark:focus:ring-pink-800
      `}        
    >
      {props.children}             
      {props.text}         
    </button>   
  );
};
  
export default Button;