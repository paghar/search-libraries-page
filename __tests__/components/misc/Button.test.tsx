import {render, fireEvent} from "@testing-library/react";
import Button from "../../../src/components/misc/Button";
import Search from "../../../src//components/misc/svg/Search";
 
describe("Button Component", () => {

  const {container} = render(
    <Button
      id="search"
      type="button"             
      onClick={()=>jest.fn()}
      text="Search"
    >
      <Search className={`mr-2 -ml-1 w-5 h-5`}/>
    </Button> 
  );

  it("Renders Snapshot Buttons", () => {    
    expect(container).toMatchSnapshot();
  });

});

describe("Button Component", () => {
  // Test rendering with required props
  test("renders button with text correctly", () => {
    const onClickMock = jest.fn();
    const {getByText} = render(
      <Button onClick={onClickMock} text="Click me" type="button" />
    );
    const buttonElement = getByText("Click me");
    expect(buttonElement).toBeInTheDocument();
  });

  // Test clicking the button
  test("calls onClick prop when button is clicked", () => {
    const onClickMock = jest.fn();
    const {getByText} = render(
      <Button onClick={onClickMock} text="Click me" type="button" />
    );
    const buttonElement = getByText("Click me");
    fireEvent.click(buttonElement);
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  // Test optional props
  test("renders button with optional id and className props", () => {
    const onClickMock = jest.fn();
    const {getByText} = render(
      <Button
        onClick={onClickMock}
        text="Click me"
        type="button"
        id="testId"
        className="`mr-2 -ml-1 w-5 h-5`"
      />
    );
    const buttonElement = getByText("Click me");
    expect(buttonElement).toHaveAttribute("id", "testId");
    expect(buttonElement).toHaveClass("inline-flex items-center");
  });

  // Test rendering with children
  test("renders button with children correctly", () => {
    const onClickMock = jest.fn();
    const {getByText} = render(
      <Button onClick={onClickMock} type="button">
        <span>Icon</span>
        <span>Click me</span>
      </Button>
    );
    expect(getByText("Icon")).toBeInTheDocument();
    expect(getByText("Click me")).toBeInTheDocument();
  });
});