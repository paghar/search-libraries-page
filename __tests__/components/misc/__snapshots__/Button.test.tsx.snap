// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Renders Snapshot Buttons 1`] = `
<div>
  <button
    class="undefined
        inline-flex items-center 
        py-2.5 px-3 ml-2 text-sm 
        font-medium text-white
        bg-pink-700 rounded-lg border border-pink-700
        hover:bg-pink-800 focus:ring-4 
        focus:outline-none focus:ring-pink-300
        dark:bg-pink-600 dark:hover:bg-pink-700 dark:focus:ring-pink-800
      "
    id="search"
    type="button"
  >
    <svg
      class="mr-2 -ml-1 w-5 h-5"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
    Search
  </button>
</div>
`;
